### PR TITLE
CB-11413 make environment service states restartable with in memory state and log context re-initialisation

### PR DESCRIPTION
**Pull request by biharitomi**
…

See detailed description in the commit message.

### CODE REVIEW

**Comment by hreeve-cloudera** [comment]
@biharitomi Could we add some information to the JIRA description about what's being done and why, so there's a little more context?

**Review by hreeve-cloudera — COMMENTED** [review]
LGTM. I'd love some more information about what problem these changes are addressing (is it an improvement or fixing a bug?), but everything looks good.

**Comment by biharitomi** [comment]
> LGTM. I'd love some more information about what problem these changes are addressing (is it an improvement or fixing a bug?), but everything looks good.

Yeah, the commit is referring to the subtask, but the parent task contains more information about the fix which is also an improvement too in some cases 😄  : https://jira.cloudera.com/browse/CB-11412

**Comment by biharitomi** [comment]
> > LGTM. I'd love some more information about what problem these changes are addressing (is it an improvement or fixing a bug?), but everything looks good.
> 
> Yeah, the commit is referring to the subtask, but the parent task contains more information about the fix which is also an improvement too in some cases 😄 : https://jira.cloudera.com/browse/CB-11412

**Comment by biharitomi** [comment]
retest this please

File: environment/src/main/java/com/sequenceiq/environment/environment/flow/EnvironmentFillInMemoryStateStoreRestartAction.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        LOGGER.debug("Restoring MDC context and InMemoryStateStore entry for flow: '{}', flow chain: '{}'", flowParameters.getFlowId(), flowChainId);
+        Payload envPayload = (Payload) payload;
+        Optional<Environment> environment = environmentService.getById(envPayload.getResourceId());
+        environment.ifPresent(env -> restart(flowParameters, flowChainId, event, payload, env));
+    }
+
+    protected void restart(FlowParameters flowParameters, String flowChainId, String event, Object payload, Environment environment) {
+        EnvironmentStatusUpdater.update(environment.getId(), environment.getStatus());
+        MDCBuilder.buildMdcContext(environment);
+        LOGGER.debug("MDC context and InMemoryStateStore entry have been restored for flow: '{}', flow chain: '{}'", flowParameters.getFlowId(), flowChainId);
```
Comment (topolyai5) [inline environment/src/main/java/com/sequenceiq/environment/environment/flow/EnvironmentFillInMemoryStateStoreRestartAction.java:37]: this is not so useful. You have already added the flowId to the MDC context. flow name, event name better and more meaningful in this log

**Review by topolyai5 — APPROVED** [review]

**Pull request merged**